recipes/nifti_clib/all/conanfile.py

```diff
@@ -0,0 +1,149 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, get, rm, rmdir
+from conan.tools.microsoft import is_msvc, is_msvc_static_runtime
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class NiftiClibConan(ConanFile):
+    name = "nifti_clib"
+    description = "C libraries for NIFTI support"
+    license = "LicenseRef-LICENSE"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/NIFTI-Imaging/nifti_clib"
+    topics = ("image")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "use_nifti2": [True, False],
+        "use_cifti": [True, False],
+        "use_fslio": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "use_nifti2": True,
+        "use_cifti": False,  # seems to be beta?
+        "use_fslio": False  # Note in CMakeLists.txt: "If OFF, The copyright of this code is questionable for inclusion with nifti."
+    }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def validate(self):
+        if is_msvc(self) and self.options.shared:
+            # not supported due to not having dllexport definitions
+            raise ConanInvalidConfiguration(f"{self.ref} does not support -o {self.ref}:shared=True with MSVC compiler.")
```

Comment:
```suggestion
            raise ConanInvalidConfiguration(f"{self.ref} does not support -o {self.ref}:shared=True with MSVC compiler.")
        if not self.options.use_nifti2 and self.options.use_cifti:
            raise ConanInvalidConfiguration(f"{self.ref} -o '&:use_cifti=True' requires -o '&:use_nifti2=True'")
```

I see a strict relation between USE_CIFTI_CODE and USE_NIFTI2_CODE:

[NIFTI-Imaging/nifti_clib@v3.0.0/CMakeLists.txt#L170](https://github.com/NIFTI-Imaging/nifti_clib/blob/v3.0.0/CMakeLists.txt#L170)

The dependency means USE_CIFTI_CODE will be available only when USE_NIFTI2_CODE is enabled.

Testing this configuration locally, it builds/installs libnifticdf.a, but not libnifti2:

```
-- Install configuration: "Release"
-- Installing: /home/conan/.conan2/p/b/nifti2b784e1dfc2bd/p/lib/libznz.a
-- Installing: /home/conan/.conan2/p/b/nifti2b784e1dfc2bd/p/include/nifti/znzlib.h
-- Installing: /home/conan/.conan2/p/b/nifti2b784e1dfc2bd/p/lib/libniftiio.a
-- Installing: /home/conan/.conan2/p/b/nifti2b784e1dfc2bd/p/include/nifti/nifti1_io.h
-- Installing: /home/conan/.conan2/p/b/nifti2b784e1dfc2bd/p/include/nifti/nifti1.h
-- Installing: /home/conan/.conan2/p/b/nifti2b784e1dfc2bd/p/bin/nifti1_tool
-- Set runtime path of "/home/conan/.conan2/p/b/nifti2b784e1dfc2bd/p/bin/nifti1_tool" to "$ORIGIN/../lib"
-- Installing: /home/conan/.conan2/p/b/nifti2b784e1dfc2bd/p/lib/libnifticdf.a
-- Installing: /home/conan/.conan2/p/b/nifti2b784e1dfc2bd/p/include/nifti/nifticdf.h
-- Installing: /home/conan/.conan2/p/b/nifti2b784e1dfc2bd/p/bin/nifti_stats
-- Set runtime path of "/home/conan/.conan2/p/b/nifti2b784e1dfc2bd/p/bin/nifti_stats" to "$ORIGIN/../lib"
-- Installing: /home/conan/.conan2/p/b/nifti2b784e1dfc2bd/p/share/cmake/NIFTI/NIFTITargets.cmake
-- Installing: /home/conan/.conan2/p/b/nifti2b784e1dfc2bd/p/share/cmake/NIFTI/NIFTITargets-release.cmake
-- Installing: /home/conan/.conan2/p/b/nifti2b784e1dfc2bd/p/share/cmake/NIFTI/NIFTIConfig.cmake
-- Installing: /home/conan/.conan2/p/b/nifti2b784e1dfc2bd/p/share/cmake/NIFTI/NIFTIConfigVersion.cmake
```